subprojects/files/src/main/java/org/gradle/internal/file/Deleter.java

```diff
@@ -37,6 +51,28 @@
      */
     boolean deleteRecursively(File target, boolean followSymlinks) throws IOException;
 
+    /**
+     * Makes sure that the given target is an empty directory.
+     *
+     * If target is...
+     *
+     * <ul>
+     *     <li>a directory, then its contents are removed recursively,</li>
+     *     <li>a symlink pointing to an existing directory, then the linked directory's contents are removed recursively,</li>
```

Comment:
Should this still be the case? Shouldn't the symlink be removed and an empty directory created instead? WDYT?

Reply:
Good point.